tools/clientSimulator/pfClientSimulator.py

```diff
@@ -0,0 +1,208 @@
+    consoleLogger = logging.StreamHandler()
+    if args.verbose:
+        consoleLogger.setLevel(logging.DEBUG)
+    else:
+        consoleLogger.setLevel(logging.INFO)
+    logger.addHandler(consoleLogger)
+
+    # The given directory should have a conf.json file
+    if not os.path.isfile(os.path.join(args.test_directory, "conf.json")):
+        # This error will only be logged in the terminal
+        logger.error(
+            "Cannot find configuration file : conf.json in {} directory.".format(
+                args.test_directory))
+        exit(1)
+
+    configParser = ConfigParser(
+        os.path.join(
+            args.test_directory,
+            "conf.json"),
+        args.test_directory,
+        consoleLogger)
+
+    # Always write all log in the file
+    logging.basicConfig(level=logging.DEBUG,
+                        format='%(name)-12s %(levelname)-8s %(message)s',
+                        filename=configParser["LogFile"],
+                        filemode='w')
+
+    # Parsing criterion file and classes generation
+    logger.info("Criterion analysis")
+    classFactory = CriterionClassFactory(configParser["CriterionFile"])
+    criterionClasses = classFactory.generateCriterionClasses()
+
+    # Tests Handlers Generation
+    testFactory = TestVectorFactory(
+        criterionClasses,
+        consoleLogger)
+
+    testLauncher = TestLauncher(
+        criterionClasses,
+        configParser,
+        consoleLogger)
+
+    # Initialisation
+    testLauncher.init(criterionClasses, args.verbose)
+
+    # Launching
+    try:
+        if args.interactive:
+            # Launch Interactive Mode with default criterions values
+            UserInteractor(
+                testLauncher,
+                testFactory.generateTestVector()).launchInteractiveMode()
+        else:
+            while True:
+                scenarioOptions = {
+                    scenarioNumber:
+                        (scenarioFileName,
+                         DynamicCallHelper(
+                             launchScenario,
+                             logger,
+                             consoleLogger,
+                             configParser["ActionGathererFile"],
+                             os.path.join(
+                                 configParser["ScenariosDirectory"], scenarioFileName),
+                             testFactory,
+                             testLauncher
+                         ))
+                    for scenarioNumber, scenarioFileName in enumerate(
+                        [file for file in sorted(os.listdir(
+                            configParser["ScenariosDirectory"]))])
+                }
+                if args.scenario is not None:
+                    scenarioOptions[args.scenario][1]()
+                    # Let the user choose other scenario after the one choosed by command line
+                    args.scenario = None
+                else:
+                    UserInteractor.getMenu(scenarioOptions)
```

Comment:
interactive mode let you design your scenario. it has nothing to do with those lines.

for the infinite loop, please see 250f0cb659a46e4dd24a7662d40610d0cae4a386